### DISCUSSION TITLE
figure out which version of montoya.api running in burp

### DISCUSSION THREAD

**Discussion by petrabrunner** [PortSwigger/burp-extensions-montoya-api#24] [Q&A] [answered]
Hi, 

I have a problem - it seems (as indicated via NoSuchMethodErrors when I try to run my extension) that the version of montoya.api that runs in my burp and the version of the interface-library I implement my extension against are not a match. 

And I cannot figure out how to figure out which version of montoya.api the burp-version is running. 

details: 
- thrown exception: 
    -         java.lang.NoSuchMethodError: 'burp.api.montoya.http.message.requests.HttpRequest burp.api.montoya.internal.MontoyaObjectFactory.httpRequest(java.lang.String)' [...]
- library is implemented against 'net.portswigger.burp.extensions:montoya-api:1.0.0' 
- burp --diagnostics output: 

> ---------------------------------------------------------------------------------------------------------
> SYSTEM PROPERTIES
> ---------------------------------------------------------------------------------------------------------
> com.sun.net.ssl.requireCloseNotify false
> file.encoding                    ANSI_X3.4-1968
> file.separator                   /
> flatlaf.uiScale.enabled          false
> java.awt.headless                true
> java.class.path                  /home/root/burp/sources/burpsuite_pro.jar
> java.class.version               61.0
> java.home                        /usr/lib/jvm/java-17-oracle
> java.io.tmpdir                   /tmp
> java.library.path                /usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
> java.runtime.name                Java(TM) SE Runtime Environment
> java.runtime.version             17.0.5+9-LTS-191
> java.specification.name          Java Platform API Specification
> java.specification.vendor        Oracle Corporation
> java.specification.version       17
> java.vendor                      Oracle Corporation
> java.vendor.url                  https://java.oracle.com/
> java.vendor.url.bug              https://bugreport.java.com/bugreport/
> java.version                     17.0.5
> java.version.date                2022-10-18
> java.vm.compressedOopsMode       Zero based
> java.vm.info                     mixed mode, sharing
> java.vm.name                     Java HotSpot(TM) 64-Bit Server VM
> java.vm.specification.name       Java Virtual Machine Specification
> java.vm.specification.vendor     Oracle Corporation
> java.vm.specification.version    17
> java.vm.vendor                   Oracle Corporation
> java.vm.version                  17.0.5+9-LTS-191
> jdk.debug                        release
> jdk.tls.allowUnsafeServerCertChange true
> jdk.tls.maxCertificateChainLength 30
> native.encoding                  ANSI_X3.4-1968
> org.bouncycastle.jsse.client.dh.minimumPrimeBits 1024
> org.bouncycastle.jsse.client.dh.unrestrictedGroups true
> os.arch                          amd64
> os.name                          Linux
> os.version                       5.10.102.1-microsoft-standard-WSL2
> path.separator                   :
> sun.arch.data.model              64
> sun.boot.library.path            /usr/lib/jvm/java-17-oracle/lib
> sun.cpu.endian                   little
> sun.font.fontmanager             sun.awt.X11FontManager
> sun.io.unicode.encoding          UnicodeLittle
> sun.java.command                 /home/root/burp/sources/burpsuite_pro.jar --diagnostics --project-file=/home/root/burp/init.burp --> > config-file=/home/root/burp/sources/projectOptions.json --config-file=/home/root/burp/sources/userOptions.json
> sun.java.launcher                SUN_STANDARD
> sun.jnu.encoding                 ANSI_X3.4-1968
> sun.management.compiler          HotSpot 64-Bit Tiered Compilers
> user.country                     US
> user.dir                         /
> user.home                        /root
> user.language                    en
> user.name                        root
> user.timezone                    GMT
> 
> ---------------------------------------------------------------------------------------------------------
> SYSTEM ENVIRONMENT
> ---------------------------------------------------------------------------------------------------------
> PATH                             /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
> HOSTNAME                         77c05f68f531
> TERM                             xterm
> PWD                              /
> LS_COLORS                       
> rs=0:di=01;34:ln=01;36: [..]
> .tgz=01;31:*. [...]
> HOME                             /root
> SHLVL                            1
> _                                /home/root/burp/run.sh
> 
> ---------------------------------------------------------------------------------------------------------
> SYSTEM RESOURCES
> ---------------------------------------------------------------------------------------------------------
> [..]
> ---------------------------------------------------------------------------------------------------------
> BURP PROPERTIES
> ---------------------------------------------------------------------------------------------------------
> Burp Version                     2022.12.6
> Build Number                     18414
> Product Name                     Burp Suite Professional
> Burp Browser version             109.0.5414.74
> Burp Browser binaries            /home/root/burp/sources/burpbrowser/109.0.5414.74
> Code source                      /home/root/burp/sources/burpsuite_pro.jar
> Debug ID                         ddpc0ddlquenvpoxrzz7:5xor
> JAR type                         Installer
> currenttimemillis                1673970325960
> nanotime                         87723805936093
> superuser                        true

**Comment by Hannah-PortSwigger** [comment, marked as answer]
Hi

v1.0.0 of the Montoya API is compatible with v2023.1(+). This is currently available on our Early Adopter channel.

You can find the latest releases and downloads here: https://portswigger.net/burp/releases

To change your channel, go to "Settings" and search "Channel" - you can then change over to Early Adopter. The auto-update functionality should prompt you to restart Burp once the new version has finished downloading.

**Reply by petrabrunner** [reply]
Hi, 

thx for that information. 
can you give me an outlook when the 2023.x versions are planned to go into the stable-channel as well? 

and would it be an option to also output the version of the extensions-framework into the --diagnostics -> that would make it easier for me (us users) to google if we are using a compatible version in burp and the extensions-code.

**Reply by Hannah-PortSwigger** [reply]
Hi

Releases normally stay on the Early Adopter channel for about two weeks before becoming stable. This can change depending on any issues found or the type of new functionality that we're adding.

With your feature request to include the Montoya API version in Burp - could you drop us an email at support@portswigger.net with the details, please?

**Reply by Hannah-PortSwigger** [reply]
We've changed the naming convention of the Montoya API to line up with minimum compatible Burp versions. This should make it clearer which versions are compatible.

All future versions of the Montoya API will be backward-compatible to v1.0.0/2023.1 of the Montoya API.
